snow/engine/avalanche/bootstrap/bootstrapper.go

```diff
@@ -614,39 +609,18 @@ func (b *bootstrapper) checkFinish(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	if linearized {
-		b.processedCache.Flush()
-		return b.OnFinished(ctx, b.Config.SharedCfg.RequestID)
+	if !linearized {
+		b.Ctx.Log.Debug("checking for stop vertex before finishing bootstrapping")
+		return b.Restart(ctx, true)
 	}
 
-	previouslyExecuted := b.executedStateTransitions
-	b.executedStateTransitions = executedVts
-
-	// Note that executedVts < c*previouslyExecuted is enforced so that the
-	// bootstrapping process will terminate even as new vertices are being
-	// issued.
-	if executedVts > 0 && executedVts < previouslyExecuted/2 && b.Config.RetryBootstrap {
-		b.Ctx.Log.Debug("checking for more vertices before finishing bootstrapping")
-		return b.Restart(ctx, true)
+	// Invariant: edge will only be the stop vertex after its acceptance.
+	edge := b.Manager.Edge(ctx)
+	stopVertexID := edge[0]
+	if err := b.VM.Linearize(ctx, stopVertexID); err != nil {
+		return err
 	}
 
-	// Notify the subnet that this chain is synced
-	b.Config.BootstrapTracker.Bootstrapped(b.Ctx.ChainID)
 	b.processedCache.Flush()
-
-	// If the subnet hasn't finished bootstrapping, this chain should remain
-	// syncing.
-	if !b.Config.BootstrapTracker.IsBootstrapped() {
```

Comment:
We don't need this anymore?

Reply:
Nope! This is handled in snowman bootstrapping now.